### ISSUE TITLE
Fails with newer versions of flake8

### ISSUE THREAD

**Issue by Goldziher** [python-formate/flake8-dunder-all#50]
Hi there, the code for the lib is no longer supported by flake8:

```text
There was a critical error during execution of Flake8:
plugin code for `flake8-dunder-all[DALL]` does not match ^[A-Z]{1,3}[0-9]{0,3}$
```

To reproduce, execute:

```yaml
  - repo: https://github.com/PyCQA/flake8
    rev: 6.0.0
    hooks:
      - id: flake8
        additional_dependencies:
          [
            "flake8-dunder-all",
          ]
```

**Comment by silasary** [comment]
Duplicate of #44
The fix is already committed to master, but hasn't been released to PyPi yet (see #48)

**Comment by silasary** [comment]
It's not clear.  There have been recent commits, but no interaction on issues/PRs in a long time.

**Comment by nkukard** [comment]
Yea, as I mentioned in my issue I"m very sure the community would very much appreciate a new release.

Failing that, I'd say our other option is to fork.

**Comment by nkukard** [comment]
Lets give it one more week and hope the maintainer steps up.

If not, I'm interested in a collab to spread workload and will fork in a week from now.

If anyone else is interested please let us know.

**Comment by Goldziher** [comment]
Sounds good.

I opened this in Ruff: https://github.com/charliermarsh/ruff/issues/3482

**Comment by nkukard** [comment]
> Sounds good.
> 
> I opened this in Ruff: [charliermarsh/ruff#3482](https://github.com/charliermarsh/ruff/issues/3482)

Not forgotten about this ... I was looking at PyPi and if there is not a release in 12 months we can submit a takeover request for an abandoned project.

Alternatively if anyone else has a name suggestion for the fork, please let me know... I've been thinking long and hard and just cannot come up with anything that makes sense.

**Comment by domdfcoding** [comment]
Sorry for the slow response, I've been busy with my PhD thesis recently.

@silasary, @Goldziher, @nkukard and others, can you confirm the master branch works fine with the latest Flake 8? If so I will make a release ASAP.

(I have no objection to integrating this into Ruff, though. It sounds like an excellent tool.)

**Comment by silasary** [comment]
Yes, master works perfectly

**Comment by nkukard** [comment]
> Sorry for the slow response, I've been busy with my PhD thesis recently.
> 
> @silasary, @Goldziher, @nkukard and others, can you confirm the master branch works fine with the latest Flake 8? If so I will make a release ASAP.
> 
> (I have no objection to integrating this into Ruff, though. It sounds like an excellent tool.)

Working for me too. Thanks for responding, its very much appreciated and a new release would be well received. :)

**Comment by Goldziher** [comment]
@domdfcoding cheers, can i also ask that you update from using list to using a tuple for dunders? We use your utility (not the flake8 rule) in Starlite, but we use tuples for `__all__`, so perhaps a configuration argument?

**Comment by nkukard** [comment]
> @domdfcoding cheers, can i also ask that you update from using list to using a tuple for dunders? We use your utility (not the flake8 rule) in Starlite, but we use tuples for `__all__`, so perhaps a configuration argument?

Configuration option would be best imho.

PEP8 seems to only show lists used and I'd guess that is more commonly accepted. I've never seen anyone using tuples. Nothing wrong with it though, everyone to their own ;)

**Comment by domdfcoding** [comment]
@Goldziher I've opened #51 to add a ``--use-tuples`` option

**Comment by domdfcoding** [comment]
I have released version 0.3.0 which should fix this. Please reopen if you continue to have issues.
